File: java/src/main/java/com/penapereira/patterns/builder/ConcreteBuilder.java
package com.penapereira.patterns.builder;

/** Assembles the parts into one representation and hands it out. */
public final class ConcreteBuilder implements Builder {

    private final Product product = new Product();

    @Override
    public void buildPartA() {
        product.add("PartA");
    }

    @Override
    public void buildPartB() {
        product.add("PartB");
    }

    @Override
    public Product getResult() {
        return product;
    }
}

File: typescript/src/builder/concrete-builder.ts
import type { Builder } from "./builder.ts";
import { Product } from "./product.ts";

/** Assembles the parts into one representation and hands it out. */
export class ConcreteBuilder implements Builder {
  private readonly product = new Product();

  buildPartA(): void {
    this.product.add("PartA");
  }

  buildPartB(): void {
    this.product.add("PartB");
  }

  getResult(): Product {
    return this.product;
  }
}
